### PR TITLE
server: clarify "dial kv node" vs "dial sql instance"

### PR DESCRIPTION
**Pull request by knz**
Requested by @stevendanna 
See individual two commits for details.

Epic: CRDB-28893.

### CODE REVIEW

File: pkg/server/tenant.go [hunk 1]
```diff
@@ -787,7 +787,7 @@ func (s *SQLServerWrapper) PreStart(ctx context.Context) error {
 		s.sqlCfg.ExternalIODirConfig,
 		s.sqlServer.cfg.Settings,
 		s.sqlServer.sqlIDContainer,
-		s.nodeDialer,
+		s.kvNodeDialer, // TODO(ssd): is this right?
```
Comment (stevendanna) [inline pkg/server/tenant.go:790]: Yes, I believe it is.  This is used in the external storage layer for the nodelocal service. In 23.1 we made nodelocal more clearly a KV-node service gated by a capability to use.

File: pkg/server/server.go [hunk 2]
```diff
@@ -2062,7 +2062,7 @@ func (s *topLevelServer) PreStart(ctx context.Context) error {
 		s.cfg.ExternalIODirConfig,
 		s.st,
 		s.sqlServer.sqlIDContainer,
-		s.nodeDialer,
+		s.kvNodeDialer, // TODO(ssd): is this right?
```
Comment (stevendanna) [inline pkg/server/server.go:2065]: I believe so, explained on the other instance of this.

**Review by stevendanna — COMMENTED** [review]
Thanks for this!

File: pkg/server/tenant.go (hunk 1 above)
Comment (knz) [reply pkg/server/tenant.go]: Ok, thanks. I removed the comment

**Comment by knz** [comment]
RFAL

**Review by yuzefovich — APPROVED** [review]
:lgtm:

> Reviewed 1 of 5 files at r1, 13 of 13 files at r3, 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/110251#-:-Ne4jC5115H5O6wtm9bV:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @DarrylWong, @knz, @rachitgsrivastava, @Santamaura, and @stevendanna)

___
*[`pkg/sql/distsql_running.go` line 259 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/110251#-Ne4idto25v528ucFI76:-Ne4idto25v528ucFI77:bi05xi7) ([raw file](https://github.com/cockroachdb/cockroach/blob/1d68b60e413b0742a8e3cf0d07c25c1816539553/pkg/sql/distsql_running.go#L259)):*
> ```Go
> 					}
> 					log.VEventf(parentCtx, 2, "worker %d is canceling at most %d flows on node %d", workerID, len(req.FlowIDs), sqlInstanceID)
> 					// TODO: Double check that we only ever cancel flows on SQL nodes/pods here.
> ```

nit: I think we can remove this TODO.

**Comment by knz** [comment]
> nit: I think we can remove this TODO.

Done

**Comment by knz** [comment]
bors r=stevendanna,yuzefovich

**Pull request merged**